faros_event.sh

```diff
@@ -973,6 +1111,191 @@ function addRunToData() {
+            }' <<< "$request_body"
+        )
+    fi
+    if ! [ -z "$environments" ]; then
+        request_body=$(jq \
+            --arg environments "$environments" \
+            '.data.test +=
+            {
+                "environments": $environments
+            }' <<< "$request_body"
+        )
+    fi
+    if ! [ -z "$device_name" ]; then
+        request_body=$(jq \
+            --arg device_name "$device_name" \
+            '.data.test.deviceInfo +=
+            {
+                "name": $device_name
+            }' <<< "$request_body"
+        )
+    fi
+    if ! [ -z "$device_os" ]; then
+        request_body=$(jq \
+            --arg device_os "$device_os" \
+            '.data.test.deviceInfo +=
+            {
+                "os": $device_os
+            }' <<< "$request_body"
+        )
+    fi
+    if ! [ -z "$device_browser" ]; then
+        request_body=$(jq \
+            --arg device_browser "$device_browser" \
+            '.data.test.deviceInfo +=
+            {
+                "browser": $device_browser
+            }' <<< "$request_body"
+        )
+    fi
+    if ! [ -z "$device_type" ]; then
+        request_body=$(jq \
+            --arg device_type "$device_type" \
+            '.data.test.deviceInfo +=
+            {
+                "type": $device_type
+            }' <<< "$request_body"
+        )
+    fi
+    if ! [ -z "$test_start_time" ]; then
+        request_body=$(jq \
+            --arg test_start_time "$test_start_time" \
+            '.data.test +=
+            {
+                "startTime": $test_start_time
+            }' <<< "$request_body"
+        )
+    fi
+    if ! [ -z "$test_end_time" ]; then
+        request_body=$(jq \
+            --arg test_end_time "$test_end_time" \
+            '.data.test +=
+            {
+                "endTime": $test_end_time
+            }' <<< "$request_body"
+        )
+    fi
+    if ! [ -z "$defect_task" ]; then
+        request_body=$(jq \
+            --arg defect_task "$defect_task" \
+            '.data.test +=
+            {
+                "defectTask": $defect_task
+            }' <<< "$request_body"
+        )
+    fi
+    if ! [ -z "$test_suite_task" ]; then
+        request_body=$(jq \
+            --arg test_suite_task "$test_suite_task" \
+            '.data.test +=
+            {
+                "suiteTask": $test_suite_task
+            }' <<< "$request_body"
+        )
+    fi
+    if ! [ -z "$test_execution_task" ]; then
+        request_body=$(jq \
+            --arg test_execution_task "$test_execution_task" \
+            '.data.test +=
+            {
+                "executionTask": $test_execution_task
+            }' <<< "$request_body"
+        )
+    fi
+    if ! [ -z "$task_source" ]; then
+        request_body=$(jq \
+            --arg task_source "$task_source" \
+            '.data.test +=
+            {
+                "taskSource": $task_source
+            }' <<< "$request_body"
+        )
+    fi
+}
```

Comment:
cab this whole thing be replaced with a for loop?

Reply:
Unfortunately no, each field has a different key and nesting specific to what it is.

Reply:
What exactly cannot be put into a loop argument?

Reply:
The device args are nested in the `deviceInfo` object.

Reply:
Yes, the latter. The if block can be in a loop.